python/plugins/processing/algs/qgis/HubDistanceLines.py

```diff
@@ -136,6 +136,9 @@ def processAlgorithm(self, parameters, context, feedback):
             src = f.geometry().boundingBox().center()
 
             neighbors = index.nearestNeighbor(src, 1)
+            if len(neighbors) == 0:
+                continue
```

Comment:
Okay I think I found a better solution: instead of raising an exception I continue the loop when no nearest neighbors are found.

The advantage is that an empty output layer gets returned even if the `destination hubs layer` has no features. This is the same behavior like having an empty `source points layer`.

@nyalldawson can you please have a look it this solution is better in your opinion.

Reply:
Sounds good to me!